<?xml version="1.0" ?>
<launch>
    <arg name="robot_ip" default="172.16.0.2"/>
    <arg name="load_gripper" default="true" />
    <arg name="eye_on_hand" default="false" />
    <arg name="markerSize" default="0.03" />

    <node name="virtual_joint_point_boardcaster_1" pkg="tf2_ros" type="static_transform_publisher" args="1.09751364598 0.105793852438 0.502234701461 -0.667847162926 -0.676457911387 0.219082354839 0.219972232093 world camera_link"/>

    <node name="virtual_joint_point_boardcaster_2" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 world panda_link0"/>
    
    <!-- (start your robot's MoveIt! stack, e.g. include its moveit_planning_execution.launch) -->
    <include file="$(find franka_control)/launch/franka_control.launch" >
        <arg name="robot_ip" value="$(arg robot_ip)" />
        <arg name="load_gripper" value="$(arg load_gripper)" />
    </include>
    
    <rosparam command="load" file="$(find panda_cartesian_impedance)/config/cartesian_impedance_controllers.yaml" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_ik_controller"/>
  
    <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/rviz/franka_description_with_marker.rviz"/>      
    
    <include file="$(find panda_cartesian_impedance)/launch/tracking_system.launch" >
        <arg name="markerSize" value="$(arg markerSize)" />
    </include>
    
    <!-- read object pose from marker, and command the arm !-->
    <node name="expert_pick_and_place" pkg="panda_cartesian_impedance" type="expert_pick_and_place.py" required="true" output="screen">
        <remap from="equilibrium_pose" to="cartesian_ik_controller/equilibrium_pose" />
        <param name="link_name" value="panda_link0" />  
        <param name="markers_topic" value="marker_publisher/markers" />
        <param name="robot_name" value="left_arm" /> 
        <param name="other_robot_name" value="right_arm" />
    </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />

</launch>